pkg/pmem-csi-driver/parameters/parameters.go

```diff
@@ -148,7 +151,7 @@ func Parse(origin Origin, stringmap map[string]string) (Volume, error) {
 			}
 		}
 		if !valid {
-			return result, fmt.Errorf("parameter %q invalid in this context", key)
+			return result, fmt.Errorf("%s: parameter %q invalid in this context", origin, key)
```

Comment:
I'm worried that this will lead to incomprehensible error messages for normal users. Suppose they use an invalid parameter in a storage class. Does `CreateVolumeOrigin: parameter "foobar" invalid in this context` really help them?

Changing the type and using them in logs is fine, but I'm not convinced about error messages. "This context" then refers to where the error occurred, not specifically the internal parameter origin.

Reply:
Agreed. closing.